de.fu_berlin.inf.dpp.core/src/de/fu_berlin/inf/dpp/negotiation/AbstractIncomingProjectNegotiation.java

```diff
@@ -184,8 +188,16 @@ public Status run(Map<String, IProject> projectMapping,
      *
      * @throws IOException, SarosCancellationException
      */
-    protected abstract void setup(IProgressMonitor monitor)
-        throws IOException, SarosCancellationException;
+    protected void setup(IProgressMonitor monitor) throws IOException,
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Remove the declaration of thrown exception 'java.io.IOException', as it cannot be thrown from method's body. ([squid:RedundantThrowsDeclarationCheck](https://sonarcloud.io/organizations/saros/rules#rule_key=squid%3ARedundantThrowsDeclarationCheck))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=saros_complete&pullRequest=127&issues=AWSQldZGMDLKieESChXO&open=AWSQldZGMDLKieESChXO)

[ISSUE_KEY]: AWSQldZGMDLKieESChXO